src/docs/deployment/macos.md

```diff
@@ -206,6 +207,217 @@ on TestFlight, or go ahead and release your app to the App Store.
+
+```bash
+app-store-connect fetch-signing-files YOUR.APP.BUNDLE_ID \
+    --platform MAC_OS \
+    --type MAC_APP_STORE \
+    --certificate-key=@file:/path/to/cert_key \
+    --create
+```
+
+Where `cert_key` is either your exported Mac App Distribution certificate private key
+or a new private key which will automatically generate a new certificate. 
+
+</li>
+<li markdown="1">
+
+If you do not have a Mac Installer Distribution certificate, you can create a new certificate by executing  
+
+```bash
+app-store-connect create-certificate \
+    --type MAC_INSTALLER_DISTRIBUTION \
+    --certificate-key=@file:/path/to/cert_key \
+    --save
+```
+
+with `cert_key` of the private key you created earlier.
+
+</li>
+<li markdown="1">
+
+Fetch the Mac Installer Distribution certificates:
+
+```bash
+app-store-connect list-certificates \
+    --type MAC_INSTALLER_DISTRIBUTION \
+    --certificate-key=@file:/path/to/cert_key \
+    --save
+```
+
+</li>
+<li markdown="1">
+
+Set up a new temporary keychain to be used for code signing:
+
+```bash
+keychain initialize
+```
+
+{{site.alert.warning}}
+**Restore Login Keychain!**
+After running `keychain intialize` you **must** run
+
+```bash
+keychain use-login
+```
+
+in order set your login keychain as the default to avoid potential authentication issues with apps on your machine.
+{{site.alert.end}}
+
+</li>
+<li markdown="1">
+
+Now add the fetched certificates to your keychain:
+
+```bash
+keychain add-certificates
+```
+
+</li>
+<li markdown="1">
+
+Update the Xcode project settings to use fetched code signing profiles: 
+
+```bash
+xcode-project use-profiles
+```
+
+</li>
+
+<li markdown="1">
+
+Install Flutter dependencies:
+
+```bash
+flutter packages pub get
+```
+
+</li>
+<li markdown="1">
+
+Install CocoaPods dependencies:
+
+```bash
+find . -name "Podfile" -execdir pod install \;
+```
+
+</li>
+<li markdown="1">
+
+Enable the Flutter macOS option:
+
+```bash
+flutter config --enable-macos-desktop
+```
+
+</li>
+<li markdown="1">
+
+Build the Flutter macOS project:
+
+```bash
+flutter build macos --release
+```
+
+</li>
+<li markdown="1">
+
+Package the app:
+
+```bash
+APP_NAME=$(find $(pwd) -name "*.app")
+PACKAGE_NAME=$(basename "$APP_NAME" .app).pkg
+xcrun productbuild --component "$APP_NAME" /Applications/ unsigned.pkg
+
+INSTALLER_CERT_NAME=$(keychain list-certificates \
+          | jq '.[0]
+            | select(.common_name
+            | contains("Mac Developer Installer"))
+            | .common_name' \
+          | xargs)
+xcrun productsign --sign "$INSTALLER_CERT_NAME" unsigned.pkg "$PACKAGE_NAME"
+rm -f unsigned.pkg 
+```
+
+</li>
+<li markdown="1">
+
+Publish the packaged app to App Store Connect:
+
+```bash
+app-store-connect publish \
+    --path "$PACKAGE_NAME"
+```
+
+</li>
+<li markdown="1">
+
+As mentioned earlier, don't forget to set your login keychain as the default to avoid authentication issues
+with apps on your machine:
```

Comment:
```suggestion
As mentioned earlier, don't forget to set your login keychain
as the default to avoid authentication issues
with apps on your machine:
```